src/files.js

```diff
@@ -123,6 +123,28 @@ module.exports = (common) => {
           })
         })
 
+        it('BIG buffer with progress', (done) => {
+          const expectedMultihash = 'Qme79tX2bViL26vNjPsF3DP1R9rMKMvnPYJiKTTKPrXJjq'
+
+          let progCount = 0
+          let progress = 0
+          const handler = (p) => {
+            progCount += 1
+            progress = p
+          }
+
+          ipfs.files.add(bigFile, {progress: handler}, (err, res) => {
+            expect(err).to.not.exist()
+            expect(res).to.have.length(1)
+            const file = res[0]
+            expect(file.hash).to.equal(expectedMultihash)
+            expect(file.path).to.equal(file.hash)
+            expect(progCount).to.equal(58)
+            expect(progress).to.equal(57792)
```

Comment:
Is this the actual file size?

Reply:
I ran all the tests and all have passed, I think this is the actual size. I'll double check tho.

Reply:
It should test for it. `.to.equal(sizeOfTheFile)`

Reply:
🔴 @dryajov this is now the last thing to do in order to get the whole thing merged, mind checking for the right size?

Reply:
I had to switch attention to - https://github.com/ipfs/js-ipfs/pull/1036, since that would be a major missing part of the whole effort. I'll resume this as soon as I get that one squared out.